modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlZoneOptionEnum.java

```diff
@@ -40,5 +40,8 @@ public enum IgniteSqlZoneOptionEnum {
     DATA_NODES_AUTO_ADJUST_SCALE_UP,
 
     /** Data nodes scale down auto adjust timeout. */
-    DATA_NODES_AUTO_ADJUST_SCALE_DOWN
+    DATA_NODES_AUTO_ADJUST_SCALE_DOWN,
+
+    /** Data storage engine. */
+    DATA_STORAGE_ENGINE
```

Comment:
Let's add this option to `org.apache.ignite.internal.sql.engine.sql.DistributionZoneSqlDdlParserTest#createZoneNoOptions`. Besides, since IgniteSqlZoneOption.key is now an abstract identifier, no need to maintain this enum anymore, because under `org.apache.ignite.internal.sql.engine.sql` we keep entities that is part of sql AST only.

It may be moved inside or close to DdlSqlToCommandConverter for convenience though